pkg/cdi/cache.go

```diff
@@ -283,28 +284,35 @@ func (c *Cache) highestPrioritySpecDir() (string, int) {
 func (c *Cache) WriteSpec(raw *cdi.Spec, name string) error {
```

Comment:
Should this API be moved to producer package?

Reply:
Yes, `WriteSpec` is a producer function, but we can't move this to the `producer` package yet. Once we have the relevant validation logic in place (e.g. #247) we should be able to mark `WriteSpec` deprecated and / or remove it though.

Reply:
makes sense, thanks for the explanations!